torchvision/prototype/datasets/_builtin/food101.py

```diff
@@ -0,0 +1,111 @@
+import enum
+import functools
+from pathlib import Path
+from typing import Any, Tuple, List, Dict, cast, Optional, BinaryIO
+
+from torchdata.datapipes.iter import IterDataPipe, Filter, Mapper, CSVParser, JsonParser, Demultiplexer
+from torchvision.prototype.datasets.utils import Dataset, DatasetInfo, DatasetConfig, HttpResource, OnlineResource
+from torchvision.prototype.datasets.utils._internal import (
+    hint_shuffling,
+    hint_sharding,
+    path_comparator,
+    getitem,
+    INFINITE_BUFFER_SIZE,
+)
+from torchvision.prototype.features import Label, EncodedImage
+
+
+class Food101Demux(enum.IntEnum):
+    METADATA = 0
+    IMAGES = 1
+    EXTRA = 2
+
+
+class Food101(Dataset):
+    """`The Food-101 Data Set <https://data.vision.ee.ethz.ch/cvl/datasets_extra/food-101/>`_.
+
+    The Food-101 is a challenging data set of 101 food categories, with 101'000 images.
+    For each class, 250 manually reviewed test images are provided as well as 750 training images.
+    On purpose, the training images were not cleaned, and thus still contain some amount of noise.
+    This comes mostly in the form of intense colors and sometimes wrong labels. All images were
+    rescaled to have a maximum side length of 512 pixels.
+
+    TODO: Add more details?
+
+    """
```

Comment:
The docstring of this class is not visible in the documentation. All information needs to be supplied to the `DatasetInfo` in `_make_info`. You already put the homepage there and currently we don't have a field for an extra description.

```suggestion
```

Correct me if I'm wrong, but the description is copy-pasted from the old dataset, correct? Thus, we lose no work / information if we delete it here for now and re-add it later if we have support for descriptions.

Reply:
Yes, that's correct, no need to keep it, we can remove it.